src/app/utils/utils.ts

```diff
@@ -265,12 +268,30 @@ export class Utils {
           }         
         }
 
+        if (output.length > 1) {
+          var resultOutputSplitted = output[0].split(this.COMMA);
+          output.slice(1, output.length).forEach(outputArray => {
+            let tempOutputArray = outputArray.split(this.COMMA)
+            for (let j = 0; j < tempOutputArray.length; j++) {
+              if (resultOutputSplitted[j] === this.EMPTY_STRING_CSV) {
+                resultOutputSplitted[j] = tempOutputArray[j];
+              }
+            }
+          });
+          var tempSplitted: string[] = temp[0].split(this.COMMA);
+          var defaultFields: string[] = tempSplitted.slice(0, tempSplitted.length - resultOutputSplitted.length);
+          dynamicFieldsResultArray = [defaultFields.concat(resultOutputSplitted).join(this.COMMA)];
```

Comment:
Extracting content of this if block as a method with meaningful name would be pretty useful for later for quick understanding what this algorithm does.

Reply:
@gmakharat I agree with you, working on it!